internal/pkg/cli/pipeline_deploy_test.go

```diff
@@ -718,6 +722,206 @@ func TestDeployPipelineOpts_Execute(t *testing.T) {
+
+				// check if the pipeline has been deployed using a legacy naming.
+				m.deployedPipelineLister.EXPECT().ListDeployedPipelines(appName).Return([]deploy.Pipeline{}, nil)
+
+				m.deployer.EXPECT().PipelineExists(gomock.Any()).DoAndReturn(func(in *deploy.CreatePipelineInput) (bool, error) {
+					if in.IsLegacy {
+						return false, errors.New("should not be a legacy pipeline")
+					}
+					return false, nil
+				})
+
+				m.pipelineStackConfig.EXPECT().Template().Return("name: mockEnv\ntype: Environment", nil)
+
+				m.deployedPipelineLister.EXPECT().ListDeployedPipelines(appName).Return([]deploy.Pipeline{}, nil)
+				m.deployer.EXPECT().Template(gomock.Any()).Return("name: mockEnv\ntype: Environment", nil)
+
+				m.mockDiffWriter = &strings.Builder{}
+
+				m.prompt.EXPECT().Confirm(continueDeploymentPrompt, "").Return(true, nil)
+
+				// deployPipeline
+				m.deployer.EXPECT().GetAppResourcesByRegion(&app, region).Return(mockResource, nil)
+				m.prog.EXPECT().Start(fmt.Sprintf(fmtPipelineDeployStart, pipelineName)).Times(1)
+				m.deployer.EXPECT().CreatePipeline(gomock.Any(), gomock.Any()).DoAndReturn(func(in *deploy.CreatePipelineInput, _ string) error {
+					if in.IsLegacy {
+						return errors.New("should not be a legacy pipeline")
+					}
+					return nil
+				})
+				m.prog.EXPECT().Stop(log.Ssuccessf(fmtPipelineDeployComplete, pipelineName)).Times(1)
+
+			},
+		},
+		"Successfully show diff and redeploy an existing pipeline": {
+			inApp:      &app,
+			inAppName:  appName,
+			inRegion:   region,
+			inShowDiff: true,
+			callMocks: func(m deployPipelineMocks) {
+				m.prog.EXPECT().Start(fmt.Sprintf(fmtPipelineDeployResourcesStart, appName)).Times(1)
+				m.deployer.EXPECT().AddPipelineResourcesToApp(&app, region).Return(nil)
+				m.prog.EXPECT().Stop(log.Ssuccessf(fmtPipelineDeployResourcesComplete, appName)).Times(1)
+				m.ws.EXPECT().ReadPipelineManifest(pipelineManifestPath).Return(mockPipelineManifest, nil)
+				m.ws.EXPECT().Rel(pipelineManifestPath).Return(relativePath, nil)
+				m.actionCmd.EXPECT().Execute().Times(2)
+
+				// convertStages
+				m.store.EXPECT().GetEnvironment(appName, "chicken").Return(mockEnv, nil).Times(1)
+				m.store.EXPECT().GetEnvironment(appName, "wings").Return(mockEnv, nil).Times(1)
+
+				// getArtifactBuckets
+				m.deployer.EXPECT().GetRegionalAppResources(gomock.Any()).Return(mockResources, nil)
+
+				// check if the pipeline has been deployed using a legacy naming.
+				m.deployedPipelineLister.EXPECT().ListDeployedPipelines(appName).Return([]deploy.Pipeline{}, nil)
+
+				m.pipelineStackConfig.EXPECT().Template().Return("name: mockEnv\ntype: Environment", nil)
+				m.deployedPipelineLister.EXPECT().ListDeployedPipelines(appName).Return([]deploy.Pipeline{}, nil)
+				m.deployer.EXPECT().Template(gomock.Any()).Return("name: mockEnv\ntype: Environment", nil)
+
+				m.mockDiffWriter = &strings.Builder{}
```

Comment:
i think you can remove this from all the tests - looks like it's already set on 949?